levels.go

```diff
@@ -741,6 +741,7 @@ func (s *levelsController) subcompact(it y.Iterator, kr keyRange, cd compactDef,
 				vp.Decode(vs.Value)
 			}
 			builder.Add(it.Key(), vs, vp.Len)
+			vp.Fid, vp.Len, vp.Offset = 0, 0, 0
```

Comment:
Thanks for the fix. Shouldn't `vp` be made local to `addKeys()` instead?

Reply:
It could, but it won't change the logic.

Reply:
Feel free to push a commit on this PR if you'd like that downscoped.

Reply:
Hey, I am not able to push to your branch due to permission issues. 
Can you please make the following downscoping changes and then the PR should be good to go?

```git
diff --git a/levels.go b/levels.go
index 76d282a..8c27e45 100644
--- a/levels.go
+++ b/levels.go
@@ -636,7 +636,6 @@ func (s *levelsController) subcompact(it y.Iterator, kr keyRange, cd compactDef,
 
        var lastKey, skipKey []byte
        var numBuilds, numVersions int
-       var vp valuePointer
 
        addKeys := func(builder *table.Builder) {
                timeStart := time.Now()
@@ -737,11 +736,11 @@ func (s *levelsController) subcompact(it y.Iterator, kr keyRange, cd compactDef,
                                }
                        }
                        numKeys++
+                       var vp valuePointer
                        if vs.Meta&bitValuePointer > 0 {
                                vp.Decode(vs.Value)
                        }
                        builder.Add(it.Key(), vs, vp.Len)
-                       vp.Fid, vp.Len, vp.Offset = 0, 0, 0
                }
                s.kv.opt.Debugf("LOG Compact. Added %d keys. Skipped %d keys. Iteration took: %v",
                        numKeys, numSkips, time.Since(timeStart).Round(time.Millisecond))
```

Thanks

Reply:
@jarifibrahim Seems like [@raulk](https://github.com/raulk) is inactive. Hence, I created https://github.com/dgraph-io/badger/pull/1613.
Anyways, thanks again @raulk.